lib/core.ps1

```diff
@@ -606,19 +611,40 @@ set invalid=`"='
 if !args! == !invalid! ( set args= )
 powershell -noprofile -ex unrestricted `"& '$resolved_path' $arg %args%;exit `$lastexitcode`"" | out-file "$shim.cmd" -encoding ascii
 
+        warn_on_overwrite $shim $path
         "#!/bin/sh`npowershell.exe -noprofile -ex unrestricted `"$resolved_path`" $arg `"$@`"" | out-file $shim -encoding ascii
     } elseif($path -match '\.jar$') {
+        warn_on_overwrite "$shim.cmd" $path
         "@java -jar `"$resolved_path`" $arg %*" | out-file "$shim.cmd" -encoding ascii
+
+        warn_on_overwrite $shim $path
         "#!/bin/sh`njava -jar `"$resolved_path`" $arg `"$@`"" | out-file $shim -encoding ascii
+    } elseif($path -match '\.py$') {
+        warn_on_overwrite "$shim.cmd" $path
+        "@python `"$resolved_path`" $arg %*" | out-file "$shim.cmd" -encoding ascii
+
+        warn_on_overwrite $shim $path
+        "#!/bin/sh`npython `"$resolved_path`" $arg `"$@`"" | out-file $shim -encoding ascii
+    } else {
+        warn_on_overwrite "$shim.cmd" $path
+        # find path to Git's bash so that batch scripts can run bash scripts
+        $gitdir = (Get-Item (Get-Command git -ErrorAction:Stop).Source -ErrorAction:Stop).Directory.Parent
+        if ($gitdir.FullName -imatch 'mingw') {
+          $gitdir = $gitdir.Parent
+        }
+        "@$(Join-Path (Join-Path $gitdir.FullName 'bin') 'bash.exe') `"$resolved_path`" $arg %*" | out-file "$shim.cmd" -encoding ascii
+
+        warn_on_overwrite $shim $path
+        "#!/bin/sh`n`"$resolved_path`" $arg `"$@`"" | out-file $shim -encoding ascii
     }
 }
 
 function get_shim_path() {
-    $shim_path = "$(versiondir 'scoop' 'current')\supporting\shimexe\bin\shim.exe"
+    $shim_path = "$(versiondir 'scoop' 'current')\supporting\shims\kiennq\shim.exe"
     $shim_version = get_config 'shim' 'default'
     switch ($shim_version) {
         '71' { $shim_path = "$(versiondir 'scoop' 'current')\supporting\shims\71\shim.exe"; Break }
-        'kiennq' { $shim_path = "$(versiondir 'scoop' 'current')\supporting\shims\kiennq\shim.exe"; Break }
```

Comment:
Need to leave kiennq in, or users who set this, will see error messages, yes?

Reply:
Fixed